### PR TITLE
Feat: push blocks to signer set and add `/v3/blocks/upload/`

### PR DESCRIPTION
**Pull request by kantai**
This PR:

* adds /v3/blocks/upload, and relayer handling for it
* pushing of blocks from miner to signer set
* signers push these blocks to their local node
* configure the signer set test to rely on signers to push the block (by disabling p2p block push)

There should be more test coverage of `/v3/blocks/upload`, I think, but I wanted to open this PR quickly to get @jcnelson's quick impression of the `/v3/blocks/upload` implementation, because I don't want that to get in the way of any work he's been doing that may be related (or identical).

### CODE REVIEW

**Review by obycode — DISMISSED** [review]
This looks good to me.

File: stackslib/src/net/api/postblock_v3.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        let response = node
+            .with_node_state(|network, sortdb, chainstate, _mempool, _rpc_args| {
+                let mut handle_conn = sortdb.index_handle_at_tip();
+                Relayer::process_new_nakamoto_block(
+                    &network.burnchain,
+                    &sortdb,
+                    &mut handle_conn,
+                    chainstate,
+                    &block,
+                    None,
```
Comment (jcnelson) [inline stackslib/src/net/api/postblock_v3.rs:131]: Should this also wake up the chains coordinator thread?

Comment (kantai) [reply stackslib/src/net/api/postblock_v3.rs:131]: That happens in the net::relay module when the network result is handled (https://github.com/stacks-network/stacks-core/pull/4902/files#diff-a29f0a1f3db4cbe0a08f45ff998bf03fd639c1b8608c362a421639292769a51fR1988)

**Review by jcnelson — DISMISSED** [review]

**Review by obycode — DISMISSED** [review]

**Review by jcnelson — DISMISSED** [review]

File: stacks-signer/src/client/stackerdb.rs [hunk 2]
```diff
@@ -272,25 +276,26 @@ mod tests {
...
         let ack = StackerDBChunkAckData {
             accepted: true,
             reason: None,
             metadata: None,
             code: None,
         };
         let mock_server = mock_server_from_config(&config);
-        let h = spawn(move || stackerdb.send_message_with_retry(signer_message));
+        debug!("Spawning msg sender");
+        let h = spawn(move || stackerdb.send_message_with_retry(signer_message).unwrap());
```
Comment (jbencin) [inline stacks-signer/src/client/stackerdb.rs:293]: Can this have a more descriptive name than `h`?

File: stackslib/src/net/api/postblock_v3.rs [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+/// Decode the HTTP response
+impl HttpResponse for RPCPostBlockRequestHandler {
+    fn try_parse_response(
+        &self,
+        preamble: &HttpResponsePreamble,
+        body: &[u8],
+    ) -> Result<HttpResponsePayload, Error> {
+        let accepted: StacksBlockAcceptedData = parse_json(preamble, body)?;
+        Ok(HttpResponsePayload::try_from_json(accepted)?)
```
Comment (jbencin) [inline stackslib/src/net/api/postblock_v3.rs:170]: Could be simplified to

```suggestion
        HttpResponsePayload::try_from_json(accepted)
```

File: testnet/stacks-node/src/nakamoto_node/miner.rs [hunk 4]
```diff
@@ -623,11 +626,40 @@ impl BlockMinerThread {
...
+        NakamotoChainState::accept_block(
+            &chainstate_config,
+            &block,
+            &mut sortition_handle,
+            &staging_tx,
+            headers_conn,
+            reward_set,
+        )?;
+        staging_tx.commit()?;
+        Ok(())
```
Comment (jbencin) [inline testnet/stacks-node/src/nakamoto_node/miner.rs:654]: Could simplify

```suggestion
        staging_tx.commit()
```

**Review by jbencin — DISMISSED** [review]

**Review by jbencin — APPROVED** [review]

**Pull request merged**